pkg/gradle/gradle.go

```diff
@@ -74,3 +79,69 @@ func getParametersFromOptions(options *ExecuteOptions) []string {
 
 	return parameters
 }
+
+// CreateBOM generates BOM file using CycloneDX
+func createBOM(options *ExecuteOptions, utils Utils) error {
+	// check if gradle task cyclonedxBom exists
+	stdOutBuf := new(bytes.Buffer)
+	stdOut := log.Writer()
+	stdOut = io.MultiWriter(stdOut, stdOutBuf)
+	utils.Stdout(stdOut)
+	if err := utils.RunExecutable(exec, "tasks"); err != nil {
+		return fmt.Errorf("failed list gradle tasks: %w", err)
+	}
+	if strings.Contains(stdOutBuf.String(), bomTaskName) {
+		if err := utils.RunExecutable(exec, bomTaskName); err != nil {
+			return fmt.Errorf("BOM creation failed: %w", err)
+		}
+	} else {
+		if err := createInitScript(options.BuildGradlePath); err != nil {
+			return err
+		}
+		defer deleteInitScript(options.BuildGradlePath)
+		if err := utils.RunExecutable(exec, "--init-script", filepath.Join(options.BuildGradlePath, initScriptName), bomTaskName); err != nil {
+			return fmt.Errorf("BOM creation failed: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func createInitScript(buildGradlePath string) error {
+	initScript, err := os.Create(filepath.Join(buildGradlePath, initScriptName))
```

Comment:
we could use fileutils here to make it better testable ...

Reply:
Thanks. I'll fix it

Reply:
Done